src/libraries/System.Security.Cryptography.X509Certificates/tests/CertificateCreation/CrlBuilderTests.cs

```diff
@@ -0,0 +1,1548 @@
+            reader.ReadEncodedValue();
+            // signature
+            reader.ReadEncodedValue();
+            reader.ThrowIfNotEmpty();
+
+            reader = tbs;
+            // CRL Version
+            Assert.Equal(BigInteger.One, reader.ReadInteger());
+            // signature algorithm identifier
+            reader.ReadEncodedValue();
+
+            AssertExtensions.SequenceEqual(crlIssuerName.RawData, reader.ReadEncodedValue().Span);
+
+            if (thisUpdate.HasValue)
+            {
+                Assert.Equal(ToTheSecond(thisUpdate.Value), ReadX509Time(reader));
+                Assert.Equal(ToTheSecond(nextUpdate), ReadX509Time(reader));
+            }
+            else
+            {
+                // The thisUpdate value was implicit DateTimeOffset.UtcNow,
+                // but it'll still be less than nextUpdate.
+                DateTimeOffset thisUpd = ReadX509Time(reader);
+                Assert.Equal(ToTheSecond(nextUpdate), ReadX509Time(reader));
+                AssertExtensions.LessThan(thisUpd, nextUpdate);
+            }
+
+            AsnReader wrap = reader.ReadSequence(new Asn1Tag(TagClass.ContextSpecific, 0));
+            reader.ThrowIfNotEmpty();
+            AsnReader extensions = wrap.ReadSequence();
+            wrap.ThrowIfNotEmpty();
+
+            AsnReader akidExt = extensions.ReadSequence();
+            Assert.Equal("2.5.29.35", akidExt.ReadObjectIdentifier());
+
+            if (expectedAkid.Critical)
+            {
+                Assert.True(akidExt.ReadBoolean(), "AKID.Critical");
+            }
+
+            byte[] akidBytes = akidExt.ReadOctetString();
+            akidExt.ThrowIfNotEmpty();
+            Assert.Equal(expectedAkid.RawData, akidBytes);
+
+            AsnReader crlNumberExt = extensions.ReadSequence();
+            Assert.Equal("2.5.29.20", crlNumberExt.ReadObjectIdentifier());
+            byte[] crlNumberBytes = crlNumberExt.ReadOctetString();
+            crlNumberExt.ThrowIfNotEmpty();
+            reader = new AsnReader(crlNumberBytes, AsnEncodingRules.DER);
+            Assert.Equal(expectedCrlNumber, reader.ReadInteger());
+            reader.ThrowIfNotEmpty();
+
+            extensions.ThrowIfNotEmpty();
+
+            static DateTimeOffset ToTheSecond(DateTimeOffset input)
+            {
+                long totalTicks = input.Ticks;
+                long unwantedTicks = totalTicks % TimeSpan.TicksPerSecond;
+                long truncatedTicks = totalTicks - unwantedTicks;
+                return new DateTimeOffset(truncatedTicks, input.Offset);
+            }
+        }
+
+        private static DateTimeOffset ReadX509Time(AsnReader reader)
+        {
+            if (reader.PeekTag().HasSameClassAndValue(Asn1Tag.UtcTime))
+            {
+                return reader.ReadUtcTime();
+            }
+
+            return reader.ReadGeneralizedTime();
+        }
+
+        private static ReadOnlySpan<byte> BuildEmptyExpectedCrl => new byte[]
```

Comment:
Are any of the CRLs here that are use for loading tests "real world"? Can we grab a small CRL from Microsoft, or $SOMEWHEREELSE and load it? Just looking for some coverage that we can interop outside our own bubble.

Reply:
Grabbed the smallest one out of my CRL cache that had entries in it and added a test.